docs/tutorials/initializedirtyimage.py

```diff
@@ -0,0 +1,190 @@
+
+dirty_image = torch.tensor(img.copy())  # turns it into a pytorch tensor
+rml = precomposed.SimpleNet(coords=coords, nchan=dset.nchan)
+optimizer = torch.optim.SGD(
+    rml.parameters(), lr=500.0
+)  # multiple different possiple optimizers
+rml.state_dict()  # parameters of the model
+
+
+# ### Loss Function
+
+# The loss function that will be used to optimize the initial part of the image is the pixel-to-pixel L2 norm (also known as the Euclidian Norm). It calculates the loss based off of the image-plane distance between the dirty image and the state of the ImageCube in order to make the state of the ImageCube closer to the dirty image.
+#
+# The L2 Norm (Euclidian Norm) is based off of the euclidian distance function and it takes the form of: $ L_{oss} = \sqrt{\sum_{i=1}^{N} (y_i - x_i)^2}  $
+
+
+def l2norm(sky, dirt):
+    return ((torch.sum((sky - dirt) ** 2))) ** 0.5
+
+
+# ### Training Loop
+#
+# Now we will be starting a trainining loop using this loss function to optimize our parameters. There will also be a loss tracker to see how loss falls off over time.
+
+# +
+# %%time
+
+
+loss_tracker = []
+for iteration in range(50):
+
+    optimizer.zero_grad()
+
+    rml.forward()
+
+    sky_cube = rml.icube.sky_cube
+
+    # lossfunc = torch.nn.MSELoss(reduction='sum')  #the MSELoss calculates mean squared error (squared L2 norm), so we take sqrt of it
+    # loss = (lossfunc(sky_cube,dirty_image))**.5 #https://pytorch.org/docs/stable/generated/torch.nn.MSELoss.html
+
+    loss = l2norm(sky_cube, dirty_image)
+
+    loss_tracker.append(loss.item())
+    loss.backward()
+    optimizer.step()
+# torch.save(rml, "dirty_image_model.pt")
+torch.save(rml.state_dict(), "dirty_image_model.pt")
+# -
+
+# We also can save the state of the model or the entire model itself. The torch.save(model.state_dict(), PATH) will just save the state.dict() which are the model's learned parameters. The alternative will save the entire model.
+# https://pytorch.org/tutorials/beginner/saving_loading_models.html
+
+# ### Results
+
+fig, ax = plt.subplots(nrows=1)
+ax.plot(loss_tracker)
+ax.set_xlabel("iteration")
+ax.set_ylabel("loss")
+
+# The results are seen here with the state_dict() having changed and the image now more closely resembling the dirty image, though the dirty image contains negative values which this does not so they can never be completely the same.
+
+fig, ax = plt.subplots(nrows=1)
+im = ax.imshow(
+    np.squeeze(rml.icube.sky_cube.detach().numpy()),
+    origin="lower",
+    interpolation="none",
+    extent=rml.icube.coords.img_ext,
+)
+plt.colorbar(im)
+
+rml.state_dict()
+
+# ### Loading the Model
+#
+# To demonstrate the saving and loading of the model here we will be resetting the parameters of the model and then reloading them.
+
+rml = precomposed.SimpleNet(coords=coords)
+rml.state_dict()  # parameters of the model
```

Comment:
These are the *uninitialized* parameters of the model, or with the default initializations

Reply:
I redo the uninitialized parameters to show the difference between them and then reloading them, though displaying that isn't too important. Though I suppose I should change the comments to make that clearer.

Reply:
Yup, that makes sense.